src/workspace-data/upload-data/UploadData.test.tsx

```diff
@@ -77,23 +57,70 @@ jest.mock('src/libs/feature-previews', () => ({
   isFeaturePreviewEnabled: jest.fn(),
 }));
 
+jest.mock('src/components/Dropzone', () => {
+  const actual = jest.requireActual('src/components/Dropzone');
+  return {
+    ...actual,
+    __esModule: true,
+    default: jest.fn(actual.default),
+  };
+});
+
 describe('UploadData Component', () => {
+  const files: File[] = [new File(['entity:sample_id\tname'], 'test1.tsv', { type: 'text/tab-separated-values' })];
+
   beforeEach(() => {
     jest.clearAllMocks();
+
+    asMockedFn(Workspaces).mockReturnValue(
+      partial<WorkspacesAjaxContract>({
+        getTags: jest.fn().mockResolvedValue([]),
+      })
+    );
+
+    asMockedFn(Metrics).mockReturnValue(partial<MetricsContract>({ captureEvent: jest.fn() }));
+
+    asMockedFn(GoogleStorage).mockReturnValue(
+      partial<GoogleStorageContract>({
+        list: jest.fn(async (_googleProject, _bucket, _prefix) =>
+          partial<GCSListObjectsResponse>({
+            items: [partial<GCSItem>({ name: 'test1.tsv', size: '100', updated: '2023-01-01T00:00:00Z' })],
+          })
+        ),
+        listAll: jest.fn(async (_googleProject, _bucket, _prefix) => ({ items: [] })),
+        upload: jest.fn(async () => undefined),
+      })
+    );
+
+    asMockedFn(Dropzone).mockImplementation((props) => {
+      const { onDropAccepted, children } = props;
+      const openUploader = async () => {
+        onDropAccepted && onDropAccepted(files, {} as DropEvent);
+      };
+      return <div>{children(partial<DropzoneState>({ dragging: false, openUploader, ...{} }))}</div>;
+    });
   });
 
   const selectOptionAndGoToNext = async (option: any) => {
     (useRoute as jest.Mock).mockReturnValue({
       query: { ...option },
     });
 
-    await act(async () => {
+    await waitFor(() => {
       renderWithAppContexts(<UploadData />);
     });
 
     fireEvent.click(screen.getByText('Next >'));
   };
 
+  const uploadDataFileToStorage = async () => {
+    const uploadBtn = screen.getByText('Upload');
+
+    await expect(uploadBtn).toBeInTheDocument();
+
+    fireEvent.click(uploadBtn);
+  };
+
   it('Renders data uploader', () => {
```

Comment:
please add standard // Arrange, // Act, and // Assert sections to these tests to improve readability. (blank line in between sections).  For some tests that are using helpers, you can label first section as // Arrange/Act

Reply:
Done: https://github.com/DataBiosphere/terra-ui/pull/5181/commits/b36edc7b6e16eb06a6bcc458f8f56b7ee87e4919